src/server/game/Server/Packets/BattlePetPackets.h

```diff
@@ -0,0 +1,182 @@
+        {
+        public:
+            BattlePetRequestJournal(WorldPacket&& packet) : ClientPacket(CMSG_BATTLE_PET_REQUEST_JOURNAL, std::move(packet)) { }
+
+            void Read() override { }
+        };
+
+        class BattlePetUpdates final : public ServerPacket
+        {
+        public:
+            BattlePetUpdates() : ServerPacket(SMSG_BATTLE_PET_UPDATES) { }
+
+            WorldPacket const* Write() override;
+
+            std::vector<BattlePet> Pets;
+            bool PetAdded = false;
+        };
+
+        class PetBattleSlotUpdates final : public ServerPacket
+        {
+        public:
+            PetBattleSlotUpdates() : ServerPacket(SMSG_PET_BATTLE_SLOT_UPDATES) { }
+
+            WorldPacket const* Write() override;
+
+            std::vector<BattlePetSlot> Slots;
+            bool AutoSlotted = false;
+            bool NewSlot = false;
+        };
+
+        class BattlePetSetBattleSlot final : public ClientPacket
+        {
+        public:
+            BattlePetSetBattleSlot(WorldPacket&& packet) : ClientPacket(CMSG_BATTLE_PET_SET_BATTLE_SLOT, std::move(packet)) { }
+
+            void Read() override;
+
+            ObjectGuid PetGuid;
+            uint8 Slot;
+        };
+
+        class BattlePetModifyName final : public ClientPacket
+        {
+        public:
+            BattlePetModifyName(WorldPacket&& packet) : ClientPacket(CMSG_BATTLE_PET_MODIFY_NAME, std::move(packet)) { }
+
+            void Read() override;
+
+            ObjectGuid PetGuid;
+            std::string Name;
+            DeclinedName Declined;
+        };
+
+        class BattlePetDeletePet final : public ClientPacket
+        {
+        public:
+            BattlePetDeletePet(WorldPacket&& packet) : ClientPacket(CMSG_BATTLE_PET_DELETE_PET, std::move(packet)) { }
+
+            void Read() override;
+
+            ObjectGuid PetGuid;
+        };
+
+        class BattlePetSetFlags final : public ClientPacket
+        {
+        public:
+            BattlePetSetFlags(WorldPacket&& packet) : ClientPacket(CMSG_BATTLE_PET_SET_FLAGS, std::move(packet)) { }
+
+            void Read() override;
+
+            ObjectGuid PetGuid;
+            uint32 Flags;
+            uint8 ControlType;
+        };
+
+        class CageBattlePet final : public ClientPacket
+        {
+        public:
+            CageBattlePet(WorldPacket&& packet) : ClientPacket(CMSG_CAGE_BATTLE_PET, std::move(packet)) { }
+
+            void Read() override;
+
+            ObjectGuid PetGuid;
+        };
+
+        class BattlePetDeleted final : public ServerPacket
+        {
+        public:
+            BattlePetDeleted() : ServerPacket(SMSG_BATTLE_PET_DELETED) { }
+
+            WorldPacket const* Write() override;
+
+            ObjectGuid PetGuid;
+        };
+
+        class BattlePetError final : public ServerPacket
+        {
+        public:
+            BattlePetError() : ServerPacket(SMSG_BATTLE_PET_ERROR) { }
```

Comment:
Give an initial size to this packet: `ServerPacket(SMSG_BATTLE_PET_ERROR, 5) { }`